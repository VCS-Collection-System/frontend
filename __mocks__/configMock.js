export const getConfig = (URL, configUrl, config) => {
  if(URL.indexOf(configUrl) > -1 ) {
    const country = URL.substring(configUrl.length);
    const countryConfig = config[country] || null;
    return Promise.resolve({data: countryConfig});
  } else {
    return Promise.reject({data: null});
  }
}