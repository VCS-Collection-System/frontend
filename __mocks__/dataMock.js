export const employeeInfo = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth":"2021-10-07",
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const attachmentObject = {
  "originalFileName":"blob",
  "size":100,
  "contentType":"image/png",
  "s3BucketName":"mockBucket",
  "s3UUID":"mockUUID"
}

export const vaxHistory = {
  "id":0,
  "employee": employeeInfo,
  "submissionDate":"2021-10-04T15:12:14.875",
  "review":{
     "reviewerEmployeeId":"z2",
     "reviewerNotes":"",
     "reviewDate":"2021-10-05T14:04:29.296",
     "reviewOutcome":"ACCEPTED",
     "rejectReason":""
  },
  "submittedBy":null,
  "autoApproved":false,
  "lastUpdatedTime":null,
  "lastUpdatedBy":null,
  "vaccineBrand":"MODERNA",
  "vaccineAdministrationDate":"2021-10-04",
  "fullVaccinatedFlag":false,
  "vaccineShotNumber":2,
  "documentType":"VaccineDocument",
  "attachment": {}
}

const taskEmployeeObject = {
  "employeeId":"mock1",
  "firstName":"Mock",
  "lastName":"Tester",
  "email":"test@mock.com",
  "agencyCode":"Department of Mock Safety",
  "agencyName":"Department of Mock Safety",
  "divisionCode":"App Mocking",
  "divisionName":null,
  "supervisor":null,
  "dateOfBirth": {
    "year": "2021",
    "monthValue": "12",
    "dayOfMonth": "16"
  },
  "alternateEmail":"mock@email.com",
  "age":0,
  "ageInMonth":0,
  "workforceId":"mock2",
  "hr":false
}

export const mockTask = {
  "task-input-data": {
    "documentList": {
      "com.redhat.vcs.model.VaccineDocumentList": {
        "documents": [ {
          "com.redhat.vcs.model.VaccineCardDocument": {
            "employee": {
              "com.redhat.vcs.model.Employee": taskEmployeeObject
            },
            "vaccineBrand": {
              "com.redhat.vcs.model.VaccineBrand": "MODERNA"
            },
            "attachment": {
              "com.redhat.vcs.model.Attachment": attachmentObject
            },
            "vaccineAdministrationDate": {
              "year": "2021",
              "monthValue": "12",
              "dayOfMonth": "16"
            },
            "lotNumber": "mockLot"
          } }
        }
      }
    }
  }
}
