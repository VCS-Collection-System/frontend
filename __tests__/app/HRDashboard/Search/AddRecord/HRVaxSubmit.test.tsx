import React from 'react';
import { HRVaxSubmit } from "@app/HRDashboard/Search/AddRecord/HRVaxSubmit";
import { act, fireEvent, render, screen } from '@testing-library/react'
import mockKeycloak from '../../../../../__mocks__/keycloakMock'
import { employeeInfo } from '../../../../../__mocks__/dataMock'
import { SUBMIT_VAX_URL } from '@app/utils/constants'
import { convertBase64ToBlob } from '@app/utils/utils';

let mockStatus;
const mockURL = SUBMIT_VAX_URL; 
jest.mock('@react-keycloak/web', () => ({
  useKeycloak: jest.fn( ()=>({keycloak: mockKeycloak}))
})) 

jest.mock("@app/utils/constants", () => ({
  KEYCLOAK_FIRST_NAME: "firstName",
  KEYCLOAK_LAST_NAME: "lastName",
  KEYCLOAK_EMAIL: "email",
  KEYCLOAK_COUNTRY: "attributes.agency[0]",
  KEYCLOAK_ID: "attributes.workforceid[0]",
  APPROVER_ROLES: ["mock-approver"],
  FEATURE_FLAGS: ["fakeFeature"],
  MAX_DOSES: {"AstraZeneca":2,"Janssen":1,"Johnson":1,"Moderna":2,"Novavax":2,"Pfizer":2},
  CONSENT_COUNTRIES: ["AUS"],
  GLOBAL_CONFIG: {"default": {
    "allowPdf": false,
    "dateOfBirth": true,
    "consentCheckbox": false,
    "covidTest": false,
    "customInput": true,
    "hrdashboard": true,
    "hrsearch": true,
    "fakeFeature": true,
    "proof": true,
    "proofOptional": false,
    "proofType": {"cdc": true, "divoc": false, "greenPass": false},
    "recovery": false,  
  }}
}))

const mockPost = jest.fn((url, payload) => {
  if(mockStatus == 500) {
    return Promise.reject({response: {status: 500}})
  }
  else if(mockStatus == 501) {
    return Promise.reject({})
  }
  else if(mockStatus == 502) {
    mockStatus = 501
    return Promise.resolve({status: mockStatus})
  }
  else if(mockStatus == 503) {
    mockStatus = 500
    return Promise.resolve({status: mockStatus})
  }
  else if(url === mockURL){
      return Promise.resolve({status: mockStatus})
  }
  else {
      return Promise.resolve({status: 500})
  }
})

jest.mock('@app/utils/axiosInterceptor', () => ({
  post: (url, payload) => mockPost(url, payload)
  }))

const mockCloseParentModal = jest.fn();
const mockVaxOneInfo = {
  "vaccineBrand": 'MODERNA',
  "vaccineAdministrationDate": '2021-09-01',
  "vaccineShotNumber": 1,
  "submittedBy": 'mockId',
  "lotNumber": 1,
  "proofType": "OTHER"
}

const mockVaxTwoInfo = {
  "vaccineBrand": 'MODERNA',
  "vaccineAdministrationDate": '2021-09-15',
  "vaccineShotNumber": 2,
  "submittedBy": 'mockId',
  "lotNumber": 2,
  "proofType": "OTHER"
}

const mockVaxProofInfo = {
  "vaccineBrand": 'MODERNA',
  "vaccineAdministrationDate": '2021-09-15',
  "vaccineShotNumber": 2,
  "submittedBy": 'mockId',
  "lotNumber": 1,
  "proofType": "CDC"
}

const base64Image = "data:image/png;base64,VEhJUyBJUyBUSEUgQU5TV0VSCg";

const mockFormDataOne = new FormData();
mockFormDataOne.append('employee', new Blob([JSON.stringify(employeeInfo)], {type: "application/json"}) );
mockFormDataOne.append('documents', new Blob([JSON.stringify(mockVaxOneInfo)], {type: "application/json"}));
mockFormDataOne.append('attachment', new Blob([]));

const mockFormDataTwo = new FormData();
mockFormDataTwo.append('employee', new Blob([JSON.stringify(employeeInfo)], {type: "application/json"}) );
mockFormDataTwo.append('documents', new Blob([JSON.stringify(mockVaxTwoInfo)], {type: "application/json"}));
mockFormDataTwo.append('attachment', new Blob([]));

const mockFormDataPDF = new FormData();
mockFormDataPDF.append('employee', new Blob([JSON.stringify(employeeInfo)], {type: "application/json"}) );
mockFormDataPDF.append('documents', new Blob([JSON.stringify(mockVaxProofInfo)], {type: "application/json"}));
mockFormDataPDF.append("attachment", new Blob([convertBase64ToBlob(base64Image)], { type: "application/pdf" }), "blob.pdf");

const mockFormDataJPG = new FormData();
mockFormDataJPG.append('employee', new Blob([JSON.stringify(employeeInfo)], {type: "application/json"}) );
mockFormDataJPG.append('documents', new Blob([JSON.stringify(mockVaxProofInfo)], {type: "application/json"}));
mockFormDataJPG.append("attachment", new Blob([convertBase64ToBlob(base64Image)], { type: "image/jpeg" }), "blob.jpg");

const mockFormDataPNG = new FormData();
mockFormDataPNG.append('employee', new Blob([JSON.stringify(employeeInfo)], {type: "application/json"}) );
mockFormDataPNG.append('documents', new Blob([JSON.stringify(mockVaxTwoInfo)], {type: "application/json"}));
mockFormDataPNG.append("attachment", new Blob([convertBase64ToBlob(base64Image)], { type: "image/png" }), "blob.png");

const mockFormDataEmpty = new FormData();
mockFormDataEmpty.append('employee', new Blob([JSON.stringify(employeeInfo)], {type: "application/json"}) );
mockFormDataEmpty.append('documents', new Blob([JSON.stringify(mockVaxTwoInfo)], {type: "application/json"}));
mockFormDataEmpty.append("attachment", new Blob([convertBase64ToBlob(base64Image)], { type: "image/" }), "blob.");

beforeEach(() => {
  mockStatus = 200
  sessionStorage.setItem("searchResultInfo", JSON.stringify(employeeInfo))
})

describe('<HRVaxSubmit />', ()=> {
  it('should be defined', () => {
    expect(HRVaxSubmit).toBeDefined()
  })

  it("should match existing snapshot", async () => {
    sessionStorage.clear()
    const { asFragment } = render(<HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    expect(asFragment()).toMatchSnapshot()
  })

  it("should allow to submit one shot of complete form", async () => {
    jest.spyOn(console,'error').mockReturnValue()
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(screen.getByText('Submitted Successfully!')).toBeTruthy()
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataOne)
    await act(async () => {
      fireEvent.click(screen.getAllByRole('button', { name: /Close/i})[0])
    })
    expect(mockCloseParentModal).toBeCalled()
  })

  it("should allow to submit two shots of complete form", async () => {
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })

    const addDoseButton = screen.getByText('Add Dose') 
    await act(async () => {
      fireEvent.click(addDoseButton)
    })

    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect1 = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const covidBrandSelect2 = screen.getByRole('combobox', { name: /vaccine-vendor-2/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    await act(async () => {
      fireEvent.change(covidBrandSelect1, {target: {value: 'Moderna'}})
      fireEvent.change(covidBrandSelect2, {target: {value: 'Moderna'}})
    })

    const dateTwoInput = screen.getByRole('textbox', { name: /shot-2-date/i})
    const lotTwoInput = screen.getByRole('textbox', { name: /lot-number-2/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(dateTwoInput, {target: {value: '09-02-2021'}})
    fireEvent.change(lotTwoInput, {target: {value: '2'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(screen.getByText('Submitted Successfully!')).toBeTruthy()
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataOne)
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataTwo)
    await act(async () => {
      fireEvent.click(screen.getAllByRole('button', { name: /Close/i})[0])
    })
    expect(mockCloseParentModal).toBeCalled()
  })

  it("should allow to submit with pdf", async () => {
    window.URL.createObjectURL = jest.fn();
    sessionStorage.setItem("hr_vax_image", "data:image/png;base64,VEhJUyBJUyBUSEUgQU5TV0VSCg")
    sessionStorage.setItem("hr_vax_image_filename", "blob.pdf")
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'CDC'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(screen.getByText('Submitted Successfully!')).toBeTruthy()
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataPDF)
  })

  it("should allow to submit with jpg", async () => {
    window.URL.createObjectURL = jest.fn();
    sessionStorage.setItem("hr_vax_image", "data:image/png;base64,VEhJUyBJUyBUSEUgQU5TV0VSCg")
    sessionStorage.setItem("hr_vax_image_filename", "blob.jpg")
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'CDC'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(screen.getByText('Submitted Successfully!')).toBeTruthy()
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataJPG)
  })

  it("should allow to submit with png", async () => {
    window.URL.createObjectURL = jest.fn();
    sessionStorage.setItem("hr_vax_image", "data:image/png;base64,VEhJUyBJUyBUSEUgQU5TV0VSCg")
    sessionStorage.setItem("hr_vax_image_filename", "blob.png")
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(screen.getByText('Submitted Successfully!')).toBeTruthy()
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataPNG)
  })

  it("should allow to submit with no attachment name and no proof type", async () => {
    window.URL.createObjectURL = jest.fn();
    sessionStorage.clear();
    sessionStorage.setItem("hr_vax_image", "data:image/png;base64,VEhJUyBJUyBUSEUgQU5TV0VSCg")
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(screen.getByText('Submitted Successfully!')).toBeTruthy()
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataEmpty)
  })

  it("should not display shot one error if no response", async () => {
    mockStatus = 501
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(screen.queryByText('Submitted Successfully!')).toBeNull();
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataOne)
  })

  it("should display shot one error with response", async () => {
    mockStatus = 500
    await act(async () => {
      render( <HRVaxSubmit closeParentModal={mockCloseParentModal}/>)
    })
    const dobInput = screen.getByRole('textbox', { name: /date_of_birth/i})
    const covidBrandSelect = screen.getByRole('combobox', { name: /vaccine-vendor-1/i})
    const dateOneInput = screen.getByRole('textbox', { name: /shot-1-date/i})
    const lotOneInput = screen.getByRole('textbox', { name: /lot-number-1/i})
    const vaxType = screen.getByRole('combobox', { name: /proof-type/i})
    const submitButton = screen.getByRole('button', { name: /Submit/i})

    fireEvent.change(dobInput, {target: {value: '09-01-2021'}})
    fireEvent.change(covidBrandSelect, {target: {value: 'Moderna'}})
    fireEvent.change(dateOneInput, {target: {value: '09-01-2021'}})
    fireEvent.change(lotOneInput, {target: {value: '1'}})
    fireEvent.change(vaxType, {target: {value: 'OTHER'}})

    await act(async () => {
      fireEvent.click(submitButton)
    })
    expect(mockPost).toBeCalledTimes(1) 
    expect(mockPost).toBeCalledWith(mockURL, mockFormDataOne)
    expect(screen.queryByText('Submitted Successfully!')).toBeNull();
    expect(screen.getByText('Error Submitting Form')).toBeTruthy()
  })
})