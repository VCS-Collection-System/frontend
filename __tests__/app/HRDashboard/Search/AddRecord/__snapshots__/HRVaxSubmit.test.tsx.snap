// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HRVaxSubmit /> should match existing snapshot 1`] = `
<DocumentFragment>
  <article
    class="pf-c-card"
    data-ouia-component-id="OUIA-Generated-Card-1"
    data-ouia-component-type="PF4/Card"
    data-ouia-safe="true"
    id=""
  >
    <div
      class="pf-c-card__body"
    >
      <form
        class="pf-c-form"
        novalidate=""
      >
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-label"
          >
            <label
              class="pf-c-form__label"
              for="employee-full-name"
            >
              <span
                class="pf-c-form__label-text"
              >
                Full Name
              </span>
            </label>
             
          </div>
          <div
            class="pf-c-form__group-control"
          >
            <input
              aria-invalid="false"
              aria-label="employee-full-name"
              class="pf-c-form-control"
              disabled=""
              type="text"
              value="undefined undefined"
            />
            <div
              aria-live="polite"
              class="pf-c-form__helper-text"
              id="employee-full-name-helper"
            >
              Make sure this matches the full name of the person you are submitting for.
            </div>
          </div>
        </div>
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-label"
          >
            <label
              class="pf-c-form__label"
              for="alt-first-name"
            >
              <span
                class="pf-c-form__label-text"
              >
                Preferred First Name (if applicable)
              </span>
            </label>
             
          </div>
          <div
            class="pf-c-form__group-control"
          >
            <input
              aria-invalid="false"
              aria-label="alt-first-name"
              class="pf-c-form-control"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              class="pf-c-form__helper-text"
              id="alt-first-name-helper"
            >
              Individual's first name as displayed on vaccination record if different than above.
            </div>
          </div>
        </div>
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-label"
          >
            <label
              class="pf-c-form__label"
              for="alt-last-name"
            >
              <span
                class="pf-c-form__label-text"
              >
                Preferred Last Name (if applicable)
              </span>
            </label>
             
          </div>
          <div
            class="pf-c-form__group-control"
          >
            <input
              aria-invalid="false"
              aria-label="alt-last-name"
              class="pf-c-form-control"
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              class="pf-c-form__helper-text"
              id="alt-last-name-helper"
            >
              Individual's last name as displayed on vaccination record if different than above.
            </div>
          </div>
        </div>
        <div
          class="pf-l-grid pf-m-gutter"
        >
          <div
            class="pf-l-grid__item pf-m-12-col-on-sm pf-m-6-col-on-md"
          >
            <article
              class="pf-c-card"
              data-ouia-component-id="OUIA-Generated-Card-2"
              data-ouia-component-type="PF4/Card"
              data-ouia-safe="true"
              id=""
            >
              <div
                class="pf-c-card__title"
              >
                1st Dose or Booster 
              </div>
              <div
                class="pf-c-card__body"
              >
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="shot-1-date"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Date
                      </span>
                      <span
                        aria-hidden="true"
                        class="pf-c-form__label-required"
                      >
                         *
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <div
                      class="pf-c-date-picker"
                      style="--pf-c-date-picker__input--c-form-control--width-chars: 10;"
                    >
                      <div
                        class="pf-c-date-picker__input"
                      >
                        <div
                          class="pf-c-input-group"
                        >
                          <input
                            aria-invalid="false"
                            aria-label="shot-1-date"
                            class="pf-c-form-control"
                            placeholder="MM-DD-YYYY"
                            type="text"
                            value=""
                          />
                          <button
                            aria-label="Toggle date picker"
                            class="pf-c-button pf-m-control"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 448 512"
                              width="1em"
                            >
                              <path
                                d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
                              />
                            </svg>
                          </button>
                        </div>
                      </div>
                    </div>
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="shot-1-date-helper"
                    >
                      The date you received the dose.
                    </div>
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="vaccine-vendor-1"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Vaccine Manufacturer
                      </span>
                      <span
                        aria-hidden="true"
                        class="pf-c-form__label-required"
                      >
                         *
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <select
                      aria-invalid="false"
                      aria-label="vaccine-vendor-1"
                      class="pf-c-form-control"
                      data-ouia-component-id="OUIA-Generated-FormSelect-default-1"
                      data-ouia-component-type="PF4/FormSelect"
                      data-ouia-safe="true"
                    >
                      <option
                        class=""
                        disabled=""
                        value=""
                      >
                        Please Select
                      </option>
                      <option
                        class=""
                        value="AstraZeneca"
                      >
                        AstraZeneca
                      </option>
                      <option
                        class=""
                        value="Comirnaty"
                      >
                        Comirnaty
                      </option>
                      <option
                        class=""
                        value="Covaxin"
                      >
                        Covaxin
                      </option>
                      <option
                        class=""
                        value="Covishield"
                      >
                        Covishield
                      </option>
                      <option
                        class=""
                        value="Covovax"
                      >
                        Covovax
                      </option>
                      <option
                        class=""
                        value="Janssen"
                      >
                        Janssen
                      </option>
                      <option
                        class=""
                        value="Johnson"
                      >
                        Johnson & Johnson
                      </option>
                      <option
                        class=""
                        value="Moderna"
                      >
                        Moderna
                      </option>
                      <option
                        class=""
                        value="Novavax"
                      >
                        Novavax
                      </option>
                      <option
                        class=""
                        value="Oxford"
                      >
                        Oxford
                      </option>
                      <option
                        class=""
                        value="Pfizer"
                      >
                        Pfizer
                      </option>
                      <option
                        class=""
                        value="Sinopharm"
                      >
                        Sinopharm
                      </option>
                      <option
                        class=""
                        value="Sinovac_CoronaVac"
                      >
                        Sinovac-CoronaVac
                      </option>
                      <option
                        class=""
                        value="Spikevax"
                      >
                        Spikevax
                      </option>
                      <option
                        class=""
                        value="Sputnik"
                      >
                        Sputnik
                      </option>
                      <option
                        class=""
                        value="Vaxzevria"
                      >
                        Vaxzevria
                      </option>
                    </select>
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="vaccine-vendor-1-helper"
                    >
                      The vendor of this dose as shown on the documentation.
                    </div>
                  </div>
                </div>
                <div
                  class="pf-c-form__group"
                >
                  <div
                    class="pf-c-form__group-label"
                  >
                    <label
                      class="pf-c-form__label"
                      for="lot-number-1"
                    >
                      <span
                        class="pf-c-form__label-text"
                      >
                        Lot/Batch Number
                      </span>
                    </label>
                     
                  </div>
                  <div
                    class="pf-c-form__group-control"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="lot-number-1"
                      class="pf-c-form-control"
                      type="text"
                      value=""
                    />
                    <div
                      aria-live="polite"
                      class="pf-c-form__helper-text"
                      id="lot-number-1-helper"
                    >
                      The lot/batch number of the vaccine received as shown on the documentation.
                    </div>
                  </div>
                </div>
              </div>
            </article>
          </div>
        </div>
        <a
          class=""
          data-pf-content="true"
          data-testid="add-dose"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 448 512"
            width="1em"
          >
            <path
              d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
            />
          </svg>
           Add Dose
        </a>
        <div
          class="pf-c-form__group"
        >
          <div
            class="pf-c-form__group-control"
          >
            <div
              class="pf-l-flex"
            >
              <div
                class="pf-m-align-right"
              >
                <button
                  aria-disabled="false"
                  class="pf-c-button pf-m-primary pf-m-progress"
                  data-ouia-component-id="OUIA-Generated-Button-primary-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="submit"
                >
                  <span
                    class="pf-c-button__icon pf-m-start"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 512 512"
                      width="1em"
                    >
                      <path
                        d="M440 6.5L24 246.4c-34.4 19.9-31.1 70.8 5.7 85.9L144 379.6V464c0 46.4 59.2 65.5 86.6 28.6l43.8-59.1 111.9 46.2c5.9 2.4 12.1 3.6 18.3 3.6 8.2 0 16.3-2.1 23.6-6.2 12.8-7.2 21.6-20 23.9-34.5l59.4-387.2c6.1-40.1-36.9-68.8-71.5-48.9zM192 464v-64.6l36.6 15.1L192 464zm212.6-28.7l-153.8-63.5L391 169.5c10.7-15.5-9.5-33.5-23.7-21.2L155.8 332.6 48 288 464 48l-59.4 387.3z"
                      />
                    </svg>
                  </span>
                  Submit
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </article>
</DocumentFragment>
`;
